apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/FavoriteService.java

```diff
@@ -77,23 +77,20 @@ public Favorite addFavorite(Favorite favorite) {
 
 
   public List<Favorite> search(String userId, String appId, Pageable page) {
-    boolean isUserIdEmpty = StringUtils.isEmpty(userId);
-    boolean isAppIdEmpty = StringUtils.isEmpty(appId);
+    boolean isUserIdEmpty = !StringUtils.hasLength(userId);
+    boolean isAppIdEmpty = !StringUtils.hasLength(appId);
 
     if (isAppIdEmpty && isUserIdEmpty) {
       throw new BadRequestException("user id and app id can't be empty at the same time");
     }
 
-    if (!isUserIdEmpty) {
+    //search by userId
+    if (!isUserIdEmpty && isAppIdEmpty) {
```

Comment:
I don't quite understand the logic here, why should we combine these two if statements to one? It looks to me the logic is changed as the userId variable is also used in the last search statement `return Collections.singletonList(favoriteRepository.findByUserIdAndAppId(userId, appId));`

Reply:
Yes, you're right. I missed the logic of that last line of code. I'll fix it. Thx.